src/components/interactive/dragBoxLayer.ts

```diff
@@ -0,0 +1,237 @@
+
+          this._grabbedEdges = this._getEdges(s);
+          if (!this._yResizable) {
+            this._grabbedEdges.top = false;
+            this._grabbedEdges.bottom = false;
+          }
+          if (!this._xResizable) {
+            this._grabbedEdges.left = false;
+            this._grabbedEdges.right = false;
+          }
+
+          if (!this._grabbedEdges.top && !this._grabbedEdges.bottom &&
+              !this._grabbedEdges.left && !this._grabbedEdges.right) {
+            this.bounds({
+              topLeft: s,
+              bottomRight: s
+            });
+            this._grabbedEdges.bottom = true;
+            this._grabbedEdges.right = true;
+          }
+          var bounds = this.bounds();
+          this._initTL = bounds.topLeft;
+          this._initBR = bounds.bottomRight;
+        });
+
+        this._dragInteraction.onDrag((s: Point, e: Point) => {
+          var topLeft = this._initTL;
+          var bottomRight = this._initBR;
+
+          if (this._grabbedEdges.bottom) {
+            bottomRight.y = e.y;
+          } else if (this._grabbedEdges.top) {
+            topLeft.y = e.y;
+          }
+
+          if (this._grabbedEdges.right) {
+            bottomRight.x = e.x;
+          } else if (this._grabbedEdges.left) {
+            topLeft.x = e.x;
+          }
+
+          this.bounds({
+            topLeft: topLeft,
+            bottomRight: bottomRight
+          });
+        });
+
+        this._dragInteraction.onDragEnd((s: Point, e: Point) => {
+          if (s.x === e.x && s.y === e.y) {
+            this.boxVisible(false);
+          }
+        });
+
+        this.registerInteraction(this._dragInteraction);
+      }
+
+      protected _setup() {
+        super._setup();
+
+        var createLine = () => this._box.append("line").style({
+                                 "opacity": 0,
+                                 "stroke": "pink",
+                                 "stroke-width": this._boxEdgeWidth
+                               });
+        this._boxEdgeT = createLine().classed("drag-edge-tb", true);
+        this._boxEdgeB = createLine().classed("drag-edge-tb", true);
+        this._boxEdgeL = createLine().classed("drag-edge-lr", true);
+        this._boxEdgeR = createLine().classed("drag-edge-lr", true);
+
+        var createCorner = () => this._box.append("circle")
+                                     .attr("r", this._boxEdgeWidth / 2)
+                                     .style({
+                                       "opacity": 0,
+                                       "fill": "pink"
+                                     });
+        this._boxCornerTL = createCorner().classed("drag-corner-tl", true);
+        this._boxCornerTR = createCorner().classed("drag-corner-tr", true);
+        this._boxCornerBL = createCorner().classed("drag-corner-bl", true);
+        this._boxCornerBR = createCorner().classed("drag-corner-br", true);
+      }
+
+      private _getEdges(p: Point) {
```

Comment:
It used to return `string[]`, but it is a lot easier to remove edges from this object rather than remove a string from an array.